src/WooCommerce.php

```diff
@@ -342,17 +342,19 @@ public static function woocommerce_template_loop_product_thumbnail() {
 
       if (count($variation_ids)) {
         $placeholders = implode(',', array_fill(0, count($variation_ids), '%d'));
-        $attachment_ids = array_unique(
-          array_merge(
-            $attachment_ids,
-            $wpdb->get_col(
-              $wpdb->prepare(
-                "SELECT pm.meta_value AS attachment_id
-                FROM wp_posts p
-                INNER JOIN wp_postmeta pm ON pm.post_id = p.ID AND pm.meta_key = '_thumbnail_id'
-                WHERE p.ID IN ($placeholders)
-                ORDER BY p.menu_order ASC",
-                $variation_ids
+        $attachment_ids = array_filter(
```

Comment:
This is pulling in data and also doing things that could probably be performed in the query. Instead of using `array_unique()` in PHP, can the query be optimised somehow?

I'm not sure on how the data is structured in the DB but can it use SELECT DISTINCT? 

Would `SELECT DISTINCT pm.meta_value AS attachment_id` work?

Reply:
Yes this works.